src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -160,16 +162,8 @@ We support
             }
             else
             {
-                string rawRequirementType = configurationSection[RequirementTypeKeyword];
-
-                //
-                // If requirement type is specified, parse it and set the requirementType variable
-                if (!string.IsNullOrEmpty(rawRequirementType) && !Enum.TryParse(rawRequirementType, ignoreCase: true, out requirementType))
-                {
-                    throw new FeatureManagementException(
-                        FeatureManagementError.InvalidConfigurationSetting, 
-                        $"Invalid requirement type '{rawRequirementType}' for feature '{configurationSection.Key}'.");
-                }
+                requirementType = ParseFeatureDefinitionSectionEnum(configurationSection, nameof(RequirementType), requirementType);
```

Comment:
Type name should not be used in this way. Refactoring of c# types can happen without necessetating a symetric change in configuration schema.

Reply:
This applies for all usage of `nameof`

Reply:
Oh like in the existing code as well? We use `nameof` to resolve feature filters, should we change that too?

Reply:
It'd be a good idea